lib/matplotlib/image.py

```diff
@@ -671,6 +671,10 @@ def set_data(self, A):
             raise TypeError("Image data of dtype {} cannot be converted to "
                             "float".format(self._A.dtype))
 
+        if (self._A.ndim == 3 and self._A.shape[-1] == 1):
+            # If just one dimension assume grayscale
```

Comment:
This comment isn't correct anymore - its not assuming greyscale, its just doing a mapping into the colormap?

Reply:
as that slipped past me, thanks!